examples/plugin/maven/hello/pom.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright 2016 Google Inc. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <artifactId>parent</artifactId>
+    <groupId>com.example</groupId>
+    <version>1.0-SNAPSHOT</version>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>hello</artifactId>
+
+  <build>
+    <plugins>
+      <!-- Turn on Error Prone -->
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.5</version>
+        <configuration>
+          <compilerId>javac-with-errorprone</compilerId>
+          <forceJavacCompilerUse>true</forceJavacCompilerUse>
+          <!-- maven-compiler-plugin defaults to targeting Java 5 -->
+          <source>1.8</source>
+          <target>1.8</target>
+          <!-- add custom checks to the annotation processor classpath -->
+          <annotationProcessorPaths>
+            <path>
+              <groupId>com.example</groupId>
+              <artifactId>sample-plugin</artifactId>
```

Comment:
Maybe add a note somewhere that Maven doesn't guarantee the build order and could attempt to build hello before sample-plugin, and fail because it cannot find sample-plugin (when I say Maven is broken by design 😉)  
One workaround is to add sample-plugin as a dependency (with scope provided so it's not transitive), but that feels as bad as relying on the order of module lines in the aggregator POM.